codeFellowship/codeFellowship/src/main/java/com/example/codeFellowship/controllers/SiteUserController.java

```diff
@@ -0,0 +1,79 @@
+package com.example.codeFellowship.controllers;
+
+import com.example.codeFellowship.models.SiteUser;
+import com.example.codeFellowship.repos.SiteUserRepo;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.servlet.view.RedirectView;
+
+import java.security.Principal;
+
+// TODO: 2: CheatSheet 2: Create a user controller
+@Controller
+public class SiteUserController {
+    @Autowired
+    SiteUserRepo siteUserRepo;
+
+    @Autowired
+    PasswordEncoder passwordEncoder;
+
+    @Autowired
+    private HttpServletRequest request;
+
+    @GetMapping("/")
+    public String getHomePage(Model m, Principal p){
+        if (p != null) {
+            String username = p.getName();
+            SiteUser user = siteUserRepo.findByUsername(username);
+m.addAttribute("username", username);
```

Comment:
Here, you're only passing the `username` string to your model (`index.html`). Thymeleaf won't be able to access the user's name, bio, etc. unless you pass in all of that info somehow.